runtime/src/bank.rs

```diff
@@ -4012,6 +4012,27 @@ impl Bank {
         self.accounts_data_len.store(accounts_data_len, Release)
     }
 
+    /// Update the accounts data len by adding `delta`.  Since `delta` is signed, negative values
+    /// are allowed as the means to subtract from `accounts_data_len`.
+    fn update_accounts_data_len(&self, delta: i64) {
+        /// Mixed integer ops currently not stable, so copying the impl.
+        /// Copied from: https://github.com/a1phyr/rust/blob/47edde1086412b36e9efd6098b191ec15a2a760a/library/core/src/num/uint_macros.rs#L1039-L1048
+        fn saturating_add_signed(lhs: u64, rhs: i64) -> u64 {
+            let (res, overflow) = lhs.overflowing_add(rhs as u64);
+            if overflow == (rhs < 0) {
+                res
```

Comment:
I don't really understand this. If `overflow == true` and `rhs < 0`, wouldn't we want to return 0?

Reply:
So rhs is really an i64. When it's cast to u64, it's two's complement, so the addition in unsigned terms wraps, but that's actually the correct value for adding the negative value.

So negative rhs expects overflow, and positive rhs expects no overflow. If that isn't true, then we've hit the saturating cases.

Reply:
Ah I missed the `as u64`, thought this was `overflowing_add_signed` but I see that's nightly only. Thanks for the clarification!